### PR TITLE
feature(backend): defaulting url trigger to http if no scheme is present

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the `http` scheme for URLs that doesn't have it already

## Changes

- Updates http trigger to add scheme if not present

## Fixes

- #2374 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/e435543d25e741089e5b7f70fa123228

### CODE REVIEW

File: server/executor/trigger/http.go [hunk 1]
```diff
@@ -81,7 +82,17 @@ func (te *httpTriggerer) Trigger(ctx context.Context, test model.Test, opts *Tri
...
 	}
-	req, err := http.NewRequestWithContext(ctx, strings.ToUpper(string(tReq.Method)), tReq.URL, body)
+
+	parsedUrl, err := url.Parse(tReq.URL)
+	if err != nil {
+		return response, err
+	}
+
+	if parsedUrl.Scheme == "" {
+		parsedUrl.Scheme = "http"
```
Comment (jorgeepc) [inline server/executor/trigger/http.go:92]: Should we have a constant for this?

**Review by jorgeepc — APPROVED** [review]

File: server/executor/trigger/http.go [hunk 2]
```diff
@@ -81,7 +82,17 @@ func (te *httpTriggerer) Trigger(ctx context.Context, test model.Test, opts *Tri
...
 		body = bytes.NewBufferString(tReq.Body)
 	}
-	req, err := http.NewRequestWithContext(ctx, strings.ToUpper(string(tReq.Method)), tReq.URL, body)
+
+	parsedUrl, err := url.Parse(tReq.URL)
+	if err != nil {
+		return response, err
+	}
+
+	if parsedUrl.Scheme == "" {
```
Comment (danielbdias) [inline server/executor/trigger/http.go:91]: We need to test an edge case: if the address comes with a `:443`, should we change this scheme to `https`?

File: server/executor/trigger/http.go (hunk 1 above)
Comment (schoren) [reply server/executor/trigger/http.go:92]: I don't think a constant is needed here. it's not a value that will be referenced any other place, it's just a default used in this only context.

**Review by schoren — APPROVED** [review]

File: server/executor/trigger/http.go (hunk 2 above)
Comment (xoscar) [reply server/executor/trigger/http.go:91]: I think that's just a really weird use case, here's what postman shows
Screenshot 2023-04-14 at 10 03 19

Comment (danielbdias) [reply server/executor/trigger/http.go:91]: In that case, we can live with that situation.

**Review by danielbdias — APPROVED** [review]
